src/execution/interpreter_loop.rs

```diff
@@ -2240,6 +2340,162 @@ pub(super) fn run<H: HookSet>(
 
                         trace!("Instruction: memory.fill");
                     }
+                    // https://webassembly.github.io/spec/core/exec/instructions.html#xref-syntax-instructions-syntax-instr-table-mathsf-table-init-x-y
+                    // https://webassembly.github.io/spec/core/binary/instructions.html#table-instructions 
+                    // in binary format it seems that elemidx is first ???????
+                    TABLE_INIT => {
+                        let elem_idx = wasm.read_var_u32().unwrap_validated() as usize;
+                        let table_idx = wasm.read_var_u32().unwrap_validated() as usize;
+
+                        let mut n: u32 = stack.pop_value(ValType::NumType(NumType::I32)).into(); // size
+                        let mut s: u32 = stack.pop_value(ValType::NumType(NumType::I32)).into(); // offset
+                        let mut d: u32 = stack.pop_value(ValType::NumType(NumType::I32)).into(); // dst
+
+                        let tab = store.tables.get(table_idx).unwrap_validated();
+                        let tab_len = tab.len();
+
+                        let elem_len = if store.passive_elem_indexes.contains(&elem_idx) {
+                            store.elements.get(elem_idx).unwrap_validated().len()
+                        } else {
+                            0
+                        };
+
+                        trace!("Instruction: table.init '{}' '{}' [{} {} {}] -> []", elem_idx, table_idx, d, s, n);
+                        
+                        if (s + n) as usize > elem_len || (d + n) as usize > tab_len {
+                            return Err(RuntimeError::TableOrElementAccessOutOfBounds);
+                        }
+
+                        let elem = store.elements.get(elem_idx).unwrap_validated();
+                        while n > 0 {
+                            let val = elem.elem.get(s as usize).unwrap_validated();
+
+                            #[allow(unused_labels)]
+                            'TABLE_SET: {
+                                store.tables.get_mut(table_idx).unwrap_validated().elem[d as usize] = val.clone();
+                            }
+
+                            d = d+1;
+                            s = s+1;
+                            n = n-1;
+                        }
+
+                        
+                    }
+                    ELEM_DROP => {
+                        let elem_idx = wasm.read_var_u32().unwrap_validated() as usize;
+
+                        // WARN: i'm not sure if this is okay or not
+                        store.elements.get_mut(elem_idx).unwrap_validated().elem = vec![];
```

Comment:
```suggestion
                        store.elements.get_mut(elem_idx).unwrap_validated().elem.clear();
```

Reply:
As per [official documentation](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.clear):
> Clears the vector, removing all values.
> 
> Note that this method has no effect on the allocated capacity of the vector.

While .clear() is faster, it doesn't deallocate the memory occupied by the vector.